### PR TITLE
[GeoMechanicsApplication] Cleanup of the head extrapolation unit tests

### PR DESCRIPTION
**Pull request by rfaasse**
**📝 Description**
Minor cleanup effort for the unit tests in `test_head_extrapolation_workflow.cpp`. I did the clean-up before, but didn't want to introduce more merge conflicts for the ci/gtests branch. Now that is done, this can be merged

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp [hunk 1]
```diff
@@ -10,112 +10,81 @@
 //  Main authors:    Jonathan Nuttall
 //
 
-#include <string>
 #include <iostream>
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp]: I'm not sure why we would need to include `iostream` here. I do see that we use `std::filesystem`. I've tried to replace `iostream` by the more appropriate `filesystem` and that seems to work.
```suggestion
#include <filesystem>
```

What are your thoughts about this?

File: applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp [hunk 2]
```diff
@@ -10,112 +10,81 @@
...
+                              "tests" / "test_head_extrapolation_custom_workflow_flow";
 
-    const Kratos::KratosExecute::CriticalHeadInfo critical_head_info(0, 0, 0);
-    const Kratos::KratosExecute::CallBackFunctions call_back_functions(&flow_stubs::emptyLog,
-                                                                 &flow_stubs::emptyProgress,
-                                                                 &flow_stubs::emptyLog,
-                                                                 &flow_stubs::emptyCancel);
-
-    const int status = execute.ExecuteFlowAnalysis(workingDirectory, projectFile, critical_head_info, "", call_back_functions);
+int RunTestCase(int test_case_number)
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp]: To follow the Kratos Style Guide, please rename the parameter as follows:
```suggestion
int RunTestCase(int TestCaseNumber)
```

This suggestion also applies to the parameter of function `CompareResults`.

**Review by avdg81 — COMMENTED** [review]
Thanks for removing the duplication from this test code. It looks nice and clean to me. I have two very minor suggestions related to an `#include`d header file and to rename two parameters to comply with the Kratos Style Guide.

File: applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp [hunk 3]
```diff
@@ -10,112 +10,81 @@
...
 
-    // output_files
-    std::string original = (std::string) workingDirectory + "/test_head_extrapolate_1.orig.res";
-    std::string result = (std::string) workingDirectory + "/test_head_extrapolate_1.post.res";
+    const KratosExecute::CriticalHeadInfo  critical_head_info(0, 0, 0);
+    const KratosExecute::CallBackFunctions call_back_functions(
+        &flow_stubs::emptyLog, &flow_stubs::emptyProgress, &flow_stubs::emptyLog, &flow_stubs::emptyCancel);
 
-    KRATOS_EXPECT_TRUE(TestUtilities::CompareFiles(original, result))
+    return execute.ExecuteFlowAnalysis(workingDirectory.generic_string(), projectFile,
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp]: Tests work when return KratosExecute().ExecuteFlowAnalysis(...). What is your opinion on having line 35?

**Review by markelov208 — DISMISSED** [review]
Hi Richard, that is a very nice work. One non-blocking general comment.

File: applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp (hunk 1 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp]: Agreed, done!

File: applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp (hunk 2 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp]: Done!

File: applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp (hunk 3 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/cpp_tests/test_head_extrapolation_flow_workflow.cpp]: Nice suggestion, indeed makes it shorter and clearer in my opinion

**Review by markelov208 — APPROVED** [review]

**Pull request merged**